### PR TITLE
Implement Notifications

### PR DESCRIPTION
**Pull request by joycrwu**
## Overview

Added Expo Framework for notifications

## Changes Made

Added an endpoint to send notifications, as well as various types for the push message, error message, and notif request 

## Test Coverage

Jessie tested on frontend-- it works currently to send a notification to your own device, but more work is needed to send notifications between devices. Need an endpoint to store user tokens in the backend 

## Next Steps (delete if not applicable)

As mentioned above, will develop an endpoint to store user tokens in the backend so that notifications can be sent between devices

### CODE REVIEW

File: src/services/NotifService.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { Service } from 'typedi';
+
+import { ExpoPushMessage, PushTicket } from '../types';
+// import { uploadImage } from '../utils/Requests';
+import { Expo } from 'expo-server-sdk';
+const expoServer = new Expo({ accessToken: 'lohuh5OtdO1KS4I_kJGdE0PDtnxgV4GEpTiJAIoZ' });
```
Comment (mateow99) [inline src/services/NotifService.ts:6]: the access token looks like it should be in .env

File: src/api/controllers/NotifController.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Body, CurrentUser, Delete, Get, JsonController, Params, Post } from 'routing-controllers';
```
Comment (mateow99) [inline src/api/controllers/NotifController.ts:1]: remove unused imports

File: src/services/NotifService.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { Service } from 'typedi';
+
+import { ExpoPushMessage, PushTicket } from '../types';
+// import { uploadImage } from '../utils/Requests';
```
Comment (mateow99) [inline src/services/NotifService.ts:4]: why is there a commented out import

**Review by mateow99 — APPROVED** [review]
just left a few comments with issues to fix but otherwise looks good

**Review by snajima — CHANGES_REQUESTED** [review]
+1 to mateo's comments, make sure when u remove the Expo access token u remove the commit that exposes it too

File: src/services/NotifService.ts (hunk 1 above)
Comment (joycrwu) [reply src/services/NotifService.ts:6]: yeah it should, my bad i forgot to remove it

**Comment by snajima** [comment]
You haven't deleted the [commit](https://github.com/cuappdev/resell-backend/pull/63/commits/ad8ba3e2149ca59de5901fdcfe73099c3a99d1f9) that exposes our secret value. Also, can you name the PR with [action verb] [feature] like implement notifications + open a new issue for the next step tasks in this PR?

**Review by snajima — CHANGES_REQUESTED** [review]
You haven't deleted the [commit](https://github.com/cuappdev/resell-backend/pull/63/commits/ad8ba3e2149ca59de5901fdcfe73099c3a99d1f9) that exposes our secret value. Also, can you name the PR with [action verb] [feature] like implement notifications + open a new issue for the next step tasks in this PR?

**Review by snajima — APPROVED** [review]

**Pull request merged**